mcs/class/System/Mono.UnityTls/UnityTls.cs

```diff
@@ -0,0 +1,249 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+
+namespace Mono.Unity
+{
+    using UInt8 = Byte;
+    using Int8 = Byte;
```

Comment:
You like want `SByte` here.

Reply:
I intentionally put `Byte` here since what we get on the C# side from encoding and marshaling functions are always `byte[]`
Unitytls uses UInt8 to denote raw buffers and Int8/char for strings. Since we need to push all in and outgoing strings through `Encoding` it is easier to pass byte* along. The aliases here are just there to keep the semantic in the interface and make it more similar to the c original.

I guess I should add a comment explaining that!